src/search/move_picker.h

```diff
@@ -42,7 +43,7 @@ class MovePicker {
         m_phase = phase;
     }
 
-    template<Player player> constexpr std::optional<movegen::Move> pickNextMove(const BitBoard& board, SearchTables& searchTables, movegen::ValidMoves& moves, uint8_t ply, std::optional<movegen::Move> ttMove = std::nullopt)
+    template<Player player> constexpr std::optional<movegen::Move> pickNextMove(const BitBoard& board, SearchTables& searchTables, movegen::ValidMoves& moves, uint8_t ply, std::optional<movegen::Move> ttMove = std::nullopt, std::optional<movegen::Move> prevMove = std::nullopt)
```

Comment:
I think this method is becoming a little bloated. Wouldn't it be easier to just provided the tables, moves etc to the constructor and then let this method be a little lighter? :)